source/Detector/Amptek/SGM/AmptekCommandManagerSGM.cpp

```diff
@@ -0,0 +1,213 @@
+//	return amptekCommand(RequestCommTestEchoPacket).hex();
+//}
+
+//bool AmptekCommandManagerSGM::isAcknowledgePacket(const QString &hexString) const{
+//	if(hexString.at(0) == 'f' && hexString.at(1) == 'f')
+//		return true;
+//	return false;
+//}
+
+//bool AmptekCommandManagerSGM::isAcknowledgeErrorPacket(const QString &hexString) const{
+//	if(isAcknowledgePacket(hexString) && hexString != "ff00" && hexString != "ff01")
+//		return true;
+//	return false;
+//}
+
+AmptekCommandManagerSGM::AmptekCommandManagerSGM()
+{
+	initiateAmptekCommands();
+}
+
+void AmptekCommandManagerSGM::initiateAmptekCommands()
+{
+	//Acknowledge types
+	amptekCommands_.append( AmptekCommand(AcknowledgeOk,                            "ff00", "OK"));
+	amptekCommands_.append( AmptekCommand(AcknowledgeOKWithInterfaceSharingRequest, "ff0c", "OK,with Interface Sharing Request"));
+	amptekCommands_.append( AmptekCommand(AcknowledgeSyncError,                     "ff01", "Sync Error"));
+	amptekCommands_.append( AmptekCommand(AcknowledgePIDError,                      "ff02", "PID Error"));
+	amptekCommands_.append( AmptekCommand(AcknowledgeLENError,                      "ff03", "LEN Error"));
+	amptekCommands_.append( AmptekCommand(AcknowledgeChecksumError,                 "ff04", "Checksum Error"));
+	amptekCommands_.append( AmptekCommand(AcknowledgeBadParameter,                  "ff05", "Bad Parameter"));
+	amptekCommands_.append( AmptekCommand(AcknowledgeBadHexRecrod,                  "ff06", "Bad Hex Record"));
+	amptekCommands_.append( AmptekCommand(AcknowledgeUnrecognizedCommand,           "ff07", "Unrecognized Command"));
+	amptekCommands_.append( AmptekCommand(AcknowledgeFPGAError,                     "ff08", "FPGA Error"));
+	amptekCommands_.append( AmptekCommand(AcknowledgeCP2201NotFound,                "ff09", "CP2201 Not Found"));
+	amptekCommands_.append( AmptekCommand(AcknowledgeScopeDataNotAvailable,         "ff0a", "Scope Data Not Available"));
+	amptekCommands_.append( AmptekCommand(AcknowledgePC5NotPresent,                 "ff0b", "PC5 Not Present"));
+	amptekCommands_.append( AmptekCommand(AcknowledgeI2CError,                      "ff0e", "I2C Error"));
+
+	//Response types
+	amptekCommands_.append( AmptekCommand(ResponseRequestStatusPacket,           "8001", "Request Status Packet Response"));
+	amptekCommands_.append( AmptekCommand(Response256ChannelSpectrumPlusStatus,  "8102", "256-channel spectrum plus Status"));
+	amptekCommands_.append( AmptekCommand(Response512ChannelSpectrumPlusStatus,  "8104", "512-channel spectrum plus Status"));
+	amptekCommands_.append( AmptekCommand(Response1024ChannelSpectrumPlusStatus, "8106", "1024-channel spectrum plus Status"));
+	amptekCommands_.append( AmptekCommand(ResponseConfiguration,                 "8207", "Configuration Readback"));
+	amptekCommands_.append( AmptekCommand(ResponseCommTestEchoPacket,            "8f7f", "Comm test - Echo packet Response"));
+
+	//Request types
+	QStringList tempResponses;
```

Comment:
These lines would be easier to read as a single line that would do the same thing.

```c++
QStringList tempResponses = QStringList() << "abc" << "def";
```